designer/server/src/main/scala/pl/touk/nussknacker/ui/process/processingtype/provider/ProcessingTypeDataProvider.scala

```diff
@@ -86,112 +90,119 @@ trait ProcessingTypeDataProvider[+Data, +CombinedData] {
 
   // TODO: We should return a generic type that can produce views for users with access rights to certain categories only.
   //       Thanks to that we will be sure that no sensitive data leak
-  final def combined: CombinedData = state.getCombined()
-
-  private[processingtype] def state: ProcessingTypeDataState[Data, CombinedData]
-
-  final def mapValues[TT](fun: Data => TT): ProcessingTypeDataProvider[TT, CombinedData] =
-    new TransformingProcessingTypeDataProvider[Data, CombinedData, TT, CombinedData](this, _.mapValues(fun))
-
-  final def mapCombined[CC](fun: CombinedData => CC): ProcessingTypeDataProvider[Data, CC] =
-    new TransformingProcessingTypeDataProvider[Data, CombinedData, Data, CC](this, _.mapCombined(fun))
+  final def combined: CombinedData = state.combinedDataTry.get
+
+  final protected[provider] def setStateValueAndNotifyObservers(
```

Comment:
This method still allows access to the state outside of the critical section. 
There is the assumption that it is always called within the critical section.